### PR TITLE
register values can be less than REGSIZE

### PR DESCRIPTION
**Pull request by WalterBright**
Another problem I found when debugging smaller SROA slices. Don't know how to make a test case for it, either, but the same test is used elsewhere.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
Test?

**Comment by WalterBright** [comment]
Turns out `runnable/test42.d` is a test case.

**Comment by WalterBright** [comment]
Arggh, updating `runnable/test_cdcmp.d` to reflect the better code generation is unbelievably tedious. I disabled it for the moment to see if the rest of the tests pass.

File: test/runnable/test_cdcmp.d [hunk 1]
```diff
@@ -147,11 +147,8 @@ alias baselineCases = AliasSeq!(
     Code!(ubyte, "<", ubyte)([
         /* push   rbp                     */ 0x55,
         /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
-        /* sub    rsp,0x10                */ 0x48, 0x83, 0xec, 0x10,
-        /* mov    DWORD PTR [rbp-0x8],edi */ 0x89, 0x7d, 0xf8,
-        /* cmp    sil,BYTE PTR [rbp-0x8]  */ 0x40, 0x3a, 0x75, 0xf8,
+        0x40, 0x3A, 0xF7,               // cmp  SIL,DIL
```
Comment (Geod24) [inline test/runnable/test_cdcmp.d]: Why did you just go with another format here ? It was consistent before...

**Pull request merged**